driver/lib/commands/execute.ts

```diff
@@ -218,3 +220,53 @@ const clickElement = async (self:FlutterDriver, elementBase64: string, opts) =>
         timeout
   });
 };
+
+const commandExtension = async (
+  self: FlutterDriver,
+  commandPayload: { command: string; [key: string]: any }
+) => {
+  const { command, ...params } = commandPayload;
+  const commandMapping: {
+    [key: string]: (self: FlutterDriver, params: any) => Promise<any>
+  } = {
+    'dragAndDrop': dragAndDropCommand,
```

Comment:
Does this mean the command `driver.execute_script('flutter:commandExtension', payload)` runs only `dragAndDrop` which is implemented in the app under test by the user side? For example when the `extended_commands.dart` implementation has different name such as `dragAndDrop1`, this will not work. Then, the naming could be more specific like `dragAndDropWithCommandExtension`. The naming is long but users will be able to add more extensions in the future if needed.

e.g. `driver.execute_script('flutter:dragAndDropWithCommandExtension', payload)`


Could the example move into the README.md as a new section to explain the usage in one README.md on the top page?

Reply:
> Does this mean the command driver.execute_script('flutter:commandExtension', payload) runs only dragAndDrop which is implemented in the app under test by the user side? 

Yes, at the moment it only supports the command `dragAndDrop`, which is implemented on the app user side. However, in the future we can expand `example/dart/extended_commands.dart` with additional command classes, such as adding `hasFoundWidget` or `getWidgetItemCount `as seen in the example in this [issue](https://github.com/appium/appium-flutter-driver/issues/688)
```
  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      "hasFound": finder.hasFound,
      // "foundMany": finder.found.toList().length,
    };
```

For this reason, the interface and processing were implemented in this way to allow for extensibility:
```
 const commandMapping: {
    [key: string]: (self: FlutterDriver, params: any) => Promise<any>
  } = {
    'dragAndDrop': dragAndDropCommand,
    // way to expand
    'hasFoundWidget': hasFoundWidgetCommand,
    'getWidgetItemCount': getWidgetItemCountCommand,
    
    'commandExtension': async (self, params) =>
```
In this implementation, we have the ability to make client-side calls in two ways:

```
params = {
    "startX": start_x,
    "startY": start_y,
    "endX": "0",
    "endY": end_y,
    "duration": "15000" # minimum duration needed to perform the drag & drop is 15000ms
}

# first way with inner command calls

payload = {
    "command": "commandExtension",
    "dragAndDrop": params
}

driver.execute_script("flutter:commandExtension", payload)

# second way with directly calls

payload = {
    "dragAndDrop": params
}

driver.execute_script("flutter:commandExtension", payload)
```
The first approach might be unnecessary.

> For example when the extended_commands.dart implementation has different name such as dragAndDrop1, this will not work. Then, the naming could be more specific like dragAndDropWithCommandExtension. The naming is long but users will be able to add more extensions in the future if needed.

Yes, it makes more sense to follow your suggested approach. I will update to use the direct `flutter:dragAndDropWithCommandExtension ` and add an additional argument to the `commandExtension `function.

> Could the example move into the README.md as a new section to explain the usage in one README.md on the top page?

Yes, i will do it

Reply:
Made the necessary changes and tested, it works properly

```
[9a17ddb8][FlutterDriver@8281] Calling AppiumDriver.execute() with args: ["flutter:getCenter",["eyJmaW5kZXJUeXBlIjoiQnlWYWx1ZUtleSIsImtleVZhbHVlU3RyaW5nIjoiMSIsImtleVZhbHVlVHlwZSI6IlN0cmluZyJ9"],"9a17ddb8-feb6-4b39-868c-22eb284467d6"]
[9a17ddb8][FlutterDriver] Executing Flutter driver command 'execute'
[9a17ddb8][FlutterDriver@8281] >>> {"command":"get_offset","finderType":"ByValueKey","keyValueString":"1","keyValueType":"String","offsetType":"center"}
[9a17ddb8][FlutterDriver@8281] <<< {"isError":false,"response":{"dx":205.71428571428572,"dy":116},"type":"_extensionType","method":"ext.flutter.driver"} | previous command get_offset
[9a17ddb8][FlutterDriver@8281] Responding to client with driver.execute() result: {"dx":205.71428571428572,"dy":116}
[9a17ddb8][HTTP] <-- POST /wd/hub/session/9a17ddb8-feb6-4b39-868c-22eb284467d6/execute/sync 200 57 ms - 44 
[9a17ddb8][HTTP] --> POST /wd/hub/session/9a17ddb8-feb6-4b39-868c-22eb284467d6/execute/sync {"script":"flutter:getCenter","args":["eyJmaW5kZXJUeXBlIjoiQnlWYWx1ZUtleSIsImtleVZhbHVlU3RyaW5nIjoiMiIsImtleVZhbHVlVHlwZSI6IlN0cmluZyJ9"]}
[9a17ddb8][FlutterDriver@8281] Calling AppiumDriver.execute() with args: ["flutter:getCenter",["eyJmaW5kZXJUeXBlIjoiQnlWYWx1ZUtleSIsImtleVZhbHVlU3RyaW5nIjoiMiIsImtleVZhbHVlVHlwZSI6IlN0cmluZyJ9"],"9a17ddb8-feb6-4b39-868c-22eb284467d6"]
[9a17ddb8][FlutterDriver] Executing Flutter driver command 'execute'
[9a17ddb8][FlutterDriver@8281] >>> {"command":"get_offset","finderType":"ByValueKey","keyValueString":"2","keyValueType":"String","offsetType":"center"}
[9a17ddb8][FlutterDriver@8281] <<< {"isError":false,"response":{"dx":205.71428571428572,"dy":172},"type":"_extensionType","method":"ext.flutter.driver"} | previous command get_offset
[9a17ddb8][FlutterDriver@8281] Responding to client with driver.execute() result: {"dx":205.71428571428572,"dy":172}
[9a17ddb8][HTTP] <-- POST /wd/hub/session/9a17ddb8-feb6-4b39-868c-22eb284467d6/execute/sync 200 5 ms - 44 
[9a17ddb8][HTTP] --> POST /wd/hub/session/9a17ddb8-feb6-4b39-868c-22eb284467d6/execute/sync {"script":"flutter:dragAndDropWithCommandExtension","args":[{"startX":205.71428571428572,"startY":116,"endX":"0","endY":172,"duration":"15000"}]}
[9a17ddb8][FlutterDriver@8281] Calling AppiumDriver.execute() with args: ["flutter:dragAndDropWithCommandExtension",[{"startX":205.71428571428572,"startY":116,"endX":"0","endY":172,"duration":"15000"}],"9a17ddb8-feb6-4b39-868c-22eb284467d6"]
[9a17ddb8][FlutterDriver] Executing Flutter driver command 'execute'
[9a17ddb8][FlutterDriver@8281] Responding to client with driver.execute() result: {"isError":false,"response":{"success":true},"type":"_extensionType","method":"ext.flutter.driver"}
[9a17ddb8][HTTP] <-- POST /wd/hub/session/9a17ddb8-feb6-4b39-868c-22eb284467d6/execute/sync 200 15051 ms - 109
```